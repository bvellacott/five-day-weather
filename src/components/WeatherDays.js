import React from 'react';
import WeatherCard from './WeatherCard';

function WeatherDays() {
  return (
    <ul className="App">
      { this.props.lists.map(list => <li key={list[0].dt}><WeatherCard list={list}/></li>)}
    </ul>
  );
}

export default WeatherDays;