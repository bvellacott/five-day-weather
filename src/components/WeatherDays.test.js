import React from 'react';
import WeatherDays from './WeatherDays';
import renderer from 'react-test-renderer';

test('Location set', () => {
  const component = renderer.create(
    <WeatherDays lists={[
    [{
      "dt": 1479438000,
      "weather": [{
        "description": "light rain",
        "icon": "10n"
      }],
    }, {
      "dt": 1479448800,
      "weather": [{
        "description": "clear sky",
        "icon": "01n"
      }],
    }],
    [{
      "dt": 1479438123,
      "weather": [{
        "description": "heavy rain",
        "icon": "13n"
      }],
    }, {
      "dt": 1479448456,
      "weather": [{
        "description": "mist",
        "icon": "15n"
      }],
    }]
  ]}/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
