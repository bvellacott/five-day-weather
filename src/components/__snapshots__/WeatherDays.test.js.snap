exports[`test Location set 1`] = `
<ul
  className="App">
  <li>
    <div
      className="WeatherCard">
      <h2>
        Friday
      </h2>
      <ul>
        <li
          className="weather">
          <div>
            <h3>
              3:00 AM
            </h3>
            <img
              alt="light rain"
              src="test-file-stub" />
          </div>
        </li>
        <li
          className="weather">
          <div>
            <h3>
              6:00 AM
            </h3>
            <img
              alt="clear sky"
              src="test-file-stub" />
          </div>
        </li>
      </ul>
    </div>
  </li>
  <li>
    <div
      className="WeatherCard">
      <h2>
        Friday
      </h2>
      <ul>
        <li
          className="weather">
          <div>
            <h3>
              3:02 AM
            </h3>
            <img
              alt="heavy rain"
              src="test-file-stub" />
          </div>
        </li>
        <li
          className="weather">
          <div>
            <h3>
              5:54 AM
            </h3>
            <img
              alt="mist"
              src={undefined} />
          </div>
        </li>
      </ul>
    </div>
  </li>
</ul>
`;
